store/store.go

```diff
@@ -226,6 +226,22 @@ func (s Store) TmpFile() (*os.File, error) {
 	return ioutil.TempFile(dir, "")
 }
 
+// TmpNamedFile returns an *os.File with the specified name local to the same
+// filesystem as the Store, or any error encountered. If the file already
+// exists it will return the existing file in read/write mode with the cursor
+// at the end of the file.
+func (s Store) TmpNamedFile(name string) (*os.File, error) {
+	dir, err := s.TmpDir()
+	if err != nil {
+		return nil, err
+	}
+	fname := filepath.Join(dir, name)
+	if _, err := os.Stat(fname); os.IsNotExist(err) {
```

Comment:
What about other os.State errors?  shouldn't we fail with the stat error rather than falling through to the os.Create()?